### PR TITLE
fix: substitute moment in npm challenges

### LINKED ISSUE #42294: Moment.js is considered a legacy package and should be replaced with a recommended substitute
**Affected page**
[https://www.freecodecamp.org/learn/apis-and-microservices/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm](https://www.freecodecamp.org/learn/apis-and-microservices/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm)

The challenge states:
*"Add version "2.14.0" of the "moment" package to the dependencies field of your package.json file."* and adds, "*Note: Moment is a handy library for working with time and dates.*"

Since moment is [officially a legacy project](https://momentjs.com/docs/#/-project-status/) and students are students are just getting familiar with npm and its package system, we are teaching them something that they need to unlearn later on.

**Expected behavior**
I suggest we base the challenge on a recommended substitute for moment. There are a few recommendations by moment itself on the previously linked page.

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42294

~The challenges work as is, but it looses the interactivity. As there is only one version of freecodecamp, the challenge Manage npm Dependencies By Understanding Semantic Versioning doesn't require any changes as there isn't a different version to ask the camper to change it to.~
With the new `@freecodecamp/example` the deleted paragraph is no more an issue

### CODE REVIEW

**Comment by naomi-lgbt** [comment]
Hey @ieahleen - here's an empty example package we can use for these lessons: https://www.npmjs.com/package/@freecodecamp/example

I published a few different versions to cover the concepts in the lessons, let me know if you need any more versions.

image


__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by ieahleen** [comment]
Done, thank you @naomi-lgbt ! Now this is ready for review

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md [hunk 1]
```diff
@@ -45,17 +45,17 @@ Use the tilde (`~`) character to prefix the version of moment in your dependenci
   );
 ```
 
-"moment" version should match "~2.10.2"
+"@freecodecamp/example" version should match "~1.2.13"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md:48]:
```suggestion
`"@freecodecamp/example"` version should match `"~1.2.13"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md [hunk 2]
```diff
@@ -18,15 +18,15 @@ To allow an npm dependency to update to the latest PATCH version, you can prefix
...
 
-Use the tilde (`~`) character to prefix the version of moment in your dependencies, and allow npm to update it to any new PATCH release.
+Use the tilde (`~`) character to prefix the version of @freecodecamp/example in your dependencies, and allow npm to update it to any new PATCH release.
 
 **Note:** The version numbers themselves should not be changed.
 
 # --hints--
 
-"dependencies" should include "moment"
+"dependencies" should include "@freecodecamp/example"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md:29]:
```suggestion
`"dependencies"` should include `"@freecodecamp/example"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md [hunk 3]
```diff
@@ -18,15 +18,15 @@ To allow an npm dependency to update to the latest PATCH version, you can prefix
 
 # --instructions--
 
-In the package.json file, your current rule for how npm may upgrade moment is to use a specific version (2.10.2). But now, you want to allow the latest 2.10.x version.
+In the package.json file, your current rule for how npm may upgrade @freecodecamp/example is to use a specific version (1.2.13). But now, you want to allow the latest 1.2.x version.
 
-Use the tilde (`~`) character to prefix the version of moment in your dependencies, and allow npm to update it to any new PATCH release.
+Use the tilde (`~`) character to prefix the version of @freecodecamp/example in your dependencies, and allow npm to update it to any new PATCH release.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md:23]:
```suggestion
Use the tilde (`~`) character to prefix the version of `@freecodecamp/example` in your dependencies, and allow npm to update it to any new _patch_ release.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md [hunk 4]
```diff
@@ -18,15 +18,15 @@ To allow an npm dependency to update to the latest PATCH version, you can prefix
 
 # --instructions--
 
-In the package.json file, your current rule for how npm may upgrade moment is to use a specific version (2.10.2). But now, you want to allow the latest 2.10.x version.
+In the package.json file, your current rule for how npm may upgrade @freecodecamp/example is to use a specific version (1.2.13). But now, you want to allow the latest 1.2.x version.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-tilde-character-to-always-use-the-latest-patch-version-of-a-dependency.md:21]:
```suggestion
In the package.json file, your current rule for how npm may upgrade `@freecodecamp/example` is to use a specific version (1.2.13). But now, you want to allow the latest 1.2.x version.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ dashedName: use-the-caret-character-to-use-the-latest-minor-version-of-a-depende
 
 Similar to how the tilde we learned about in the last challenge allows npm to install the latest PATCH for a dependency, the caret (`^`) allows npm to install future updates as well. The difference is that the caret will allow both MINOR updates and PATCHes.
 
-Your current version of moment should be "~2.10.2" which allows npm to install to the latest 2.10.x version. If you were to use the caret (^) as a version prefix instead, npm would be allowed to update to any 2.x.x version.
+Your current version of @freecodecamp/example should be "~1.2.13" which allows npm to install to the latest 1.2.x version. If you were to use the caret (^) as a version prefix instead, npm would be allowed to update to any 1.x.x version.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md:13]:
```suggestion
Your current version of `@freecodecamp/example` should be "~1.2.13" which allows npm to install to the latest 1.2.x version. If you were to use the caret (^) as a version prefix instead, npm would be allowed to update to any 1.x.x version.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md [hunk 6]
```diff
@@ -20,13 +20,13 @@ This would allow updates to any 1.x.x version of the package.
 
 # --instructions--
 
-Use the caret (`^`) to prefix the version of moment in your dependencies and allow npm to update it to any new MINOR release.
+Use the caret (`^`) to prefix the version of @freecodecamp/example in your dependencies and allow npm to update it to any new MINOR release.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md:23]:
```suggestion
Use the caret (`^`) to prefix the version of `@freecodecamp/example` in your dependencies and allow npm to update it to any new MINOR release.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md [hunk 7]
```diff
@@ -20,13 +20,13 @@ This would allow updates to any 1.x.x version of the package.
...
 
 **Note:** The version numbers themselves should not be changed.
 
 # --hints--
 
-"dependencies" should include "moment"
+"dependencies" should include "@freecodecamp/example"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md:29]:
```suggestion
`"dependencies"` should include `"@freecodecamp/example"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md [hunk 8]
```diff
@@ -45,17 +45,17 @@ Use the caret (`^`) to prefix the version of moment in your dependencies and all
   );
 ```
 
-"moment" version should match "^2.x.x"
+"@freecodecamp/example" version should match "^1.x.x"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/use-the-caret-character-to-use-the-latest-minor-version-of-a-dependency.md:48]:
```suggestion
`"@freecodecamp/example"` version should match `"^1.x.x"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/remove-a-package-from-your-dependencies.md [hunk 9]
```diff
@@ -16,13 +16,13 @@ This same method applies to removing other fields in your package.json as well
 
 # --instructions--
 
-Remove the moment package from your dependencies.
+Remove the @freecodecamp/example package from your dependencies.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/remove-a-package-from-your-dependencies.md:19]:
```suggestion
Remove the `@freecodecamp/example` package from your dependencies.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/remove-a-package-from-your-dependencies.md [hunk 10]
```diff
@@ -16,13 +16,13 @@ This same method applies to removing other fields in your package.json as well
...
 
 **Note:** Make sure you have the right amount of commas after removing it.
 
 # --hints--
 
-"dependencies" should not include "moment"
+"dependencies" should not include "@freecodecamp/example"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/remove-a-package-from-your-dependencies.md:25]:
```suggestion
`"dependencies"` should not include `"@freecodecamp/example"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/manage-npm-dependencies-by-understanding-semantic-versioning.md [hunk 11]
```diff
@@ -20,11 +20,11 @@ The MAJOR version should increment when you make incompatible API changes. The M
 
 # --instructions--
 
-In the dependencies section of your package.json file, change the `version` of moment to match MAJOR version 2, MINOR version 10 and PATCH version 2
+In the dependencies section of your package.json file, change the `version` of @freecodecamp/example to match MAJOR version 1, MINOR version 2 and PATCH version 13
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/manage-npm-dependencies-by-understanding-semantic-versioning.md:23]:
```suggestion
In the dependencies section of your `package.json` file, change the version of `@freecodecamp/example` to match MAJOR version 1, MINOR version 2 and PATCH version 13
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/manage-npm-dependencies-by-understanding-semantic-versioning.md [hunk 12]
```diff
@@ -20,11 +20,11 @@ The MAJOR version should increment when you make incompatible API changes. The M
...
 
 # --hints--
 
-"dependencies" should include "moment"
+"dependencies" should include "@freecodecamp/example"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/manage-npm-dependencies-by-understanding-semantic-versioning.md:27]:
```suggestion
`"dependencies"` should include `"@freecodecamp/example"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/manage-npm-dependencies-by-understanding-semantic-versioning.md [hunk 13]
```diff
@@ -43,17 +43,17 @@ In the dependencies section of your package.json file, change the `version` of m
   );
 ```
 
-"moment" version should be "2.10.2"
+"@freecodecamp/example" version should be "1.2.13"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/manage-npm-dependencies-by-understanding-semantic-versioning.md:46]:
```suggestion
`"@freecodecamp/example"` version should be `"1.2.13"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md [hunk 14]
```diff
@@ -22,13 +22,13 @@ In this section, packages your project requires are stored using the following f
 
 # --instructions--
 
-Add version "2.14.0" of the "moment" package to the `dependencies` field of your package.json file.
+Add version "1.1.0" of the "@freecodecamp/example" package to the `dependencies` field of your package.json file.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md:25]:
```suggestion
Add version "1.1.0" of the `@freecodecamp/example` package to the `dependencies` field of your `package.json` file.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md [hunk 15]
```diff
@@ -22,13 +22,13 @@ In this section, packages your project requires are stored using the following f
...
 
-**Note:** Moment is a handy library for working with time and dates.
+**Note:** @freecodecamp/example is a faux package used as a learning tool.
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md:27]:
```suggestion
**Note:** `@freecodecamp/example` is a faux package used as a learning tool.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md [hunk 16]
```diff
@@ -22,13 +22,13 @@ In this section, packages your project requires are stored using the following f
...
 
 # --hints--
 
-"dependencies" should include "moment"
+"dependencies" should include "@freecodecamp/example"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md:31]:
```suggestion
`"dependencies"` should include `"@freecodecamp/example"`.
```

File: curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md [hunk 17]
```diff
@@ -47,17 +47,17 @@ Add version "2.14.0" of the "moment" package to the `dependencies` field of your
   );
 ```
 
-"moment" version should be "2.14.0"
+"@freecodecamp/example" version should be "1.1.0"
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/05-back-end-development-and-apis/managing-packages-with-npm/expand-your-project-with-external-packages-from-npm.md:50]:
```suggestion
`"@freecodecamp/example"` version should be `"1.1.0"`.
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
While we are here, can we update the formatting to be consistent with majority of the curriculum:

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Review by Sembauke — APPROVED** [review]

**Pull request merged**